pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -64,35 +68,103 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 	}
 	apiExport, err := r.getAPIExport(apiExportPath, apiExportEndpointSlice.Spec.APIExport.Name)
 	if err != nil {
+		reason := apisv1alpha1.InternalErrorReason
```

Comment:
is this an internal error so that the conditions stay stable?

Reply:
how will we know the error?

Reply:
It is an internal error not to leak any information to the user(discussed in the previous PR).
The error details are available in the logs for a privileged user.

Reply:
an error could be used for troubleshooting

Reply:
hence in the logs